monai/config/print_dependencies.py

```
@@ -0,0 +1,67 @@
+
+    proj = data[PROJ_KEY]
+    opts = proj[OPTS_KEY]
+    dependencies = list(proj[DEP_KEY])
+    sections = set(sections or [])
+
+    if BUILD_SYSTEM_KEY in sections:
+        sections.remove(BUILD_SYSTEM_KEY)
+        dependencies += data[BUILD_SYSTEM_KEY][REQ_KEY]
+
+    if "*" in sections:
+        dependencies += sum(opts.values(), [])
```

Ruff (0.15.21)

[warning] Avoid quadratic list summation

Replace with `functools.reduce`

(RUF017)